docs/rm/snax_cluster.md

```diff
@@ -0,0 +1,198 @@
+
+#### Response Channels
+
+| Signal            | Description                                                                                                   | 
+| ----------------- | ------------------------------------------------------------------------------------------------------------  |
+| id                | Transaction ID that is equal to the destination register specified in the requested custom or CSR instruction.|
+| error             | Error signal.                                                                                                 |
+| data              | The return value of the requested transaction.                                                                |
+| snax_pvalid       | Response valid signal.                                                                                        |
+| snax_pready       | Response ready signal.                                                                                        |
+
+### CSR Instruction and Accelerator Port Mapping
+
+It is important to know how to map the CSR instruction into the request and response channel ports. For example, consider the `csrrw` instruction:
+
+```asm
+csrrw rd, csr_addr, rs1
+```
+
+Where `csrrw` is a CSR read-write instruction which reads the current data located at `csr_addr` and store it in `rd` and write the data `rs1` at `csr_addr` simultanesouly. The `csr_addr` is a 12-bit unsigned value (e.g., `0x3c0`). This instruction maps to the request channel as:
+
+* `addr` - will point to the accelerator port (e.g., 0 is for the FPSS, 1 is for the shared multiply and divide, and 5 is for SNAX ports. Always use 5 if you want to use the SNAX accelerator ports.).
+* `id` - will be the register number `rd` (not the contents of `rd`).
+* `data_op` - is the machine code for the CSR instruction (e.g., the machine code equivalent of `csrrw rd, csr_addr, rs1`).
+* `data_arga` - will contain `csr_addr`.
+* `data_argb` - will contain the contents of `rs1`.
+* `data_argc` - not used for SNAX but used in SSR.
+
+Then for the response ports we have:
+
+* `id` - this indicates which destination register should the `data` be stored. This is equivalent to the requested `rd`.
+* `error` - this is custom to the accelerator.
+* `data` - the data returned by the request.
+
+### Timing Diagram for Request and Response
+
+The section [Request and Response](reqrsp_interface.md) ports discuss the valid-ready handshake. It is used for the request and response channels. Below are some sample timing diagrams for the request and response ports.
+
+![SNAX accelerator ports](https://drive.google.com/uc?id=1ORWy6HhMxf3Q21EetJjOpklhCiqumZyX)
+
+In the figure above, observe the following:
+
+* The transaction only progresses when `valid` and `ready` signals for each `request` and `response` ports are both high. The transaction progresses, otherwise, cycles are stalled.
```

Comment:
Shall we clarify that the ready signal only asserts when the valid signal goes high and the module is actually ready for operation? So if the ready signal isn't asserted, the request module needs to keep sending the valid signal. Otherwise, people may misunderstand.

Reply:
This is okay, it's described in the request and response documentation or page. Just to avoid double describing 😆